### PR TITLE
Fix Code Actions not applied when LSP editor is enabled

### PR DESCRIPTION
**Pull request by akhera99**
fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1922913

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs [hunk 1]
```diff
@@ -60,13 +60,15 @@ internal static class CodeActionHelpers
                     {
                         if (!IsCodeActionNotSupportedByLSP(suggestedAction))
                         {
+                            using var _1 = ArrayBuilder<string>.GetInstance(out var codeActionPath);
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:63]:
```suggestion
                            using var _1 = ArrayBuilder<string>.GetInstance(out var codeActionPathList);
```

(let's name things uniformly everywhere).

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs [hunk 2]
```diff
@@ -222,22 +224,15 @@ private static bool IsCodeActionNotSupportedByLSP(IUnifiedSuggestedAction sugges
             CodeActionPriority setPriority,
             LSP.Range? applicableRange,
             int currentSetNumber,
-            ref int currentHighestSetNumber,
-            string currentTitle = "")
+            ArrayBuilder<string> codeActionPath,
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:227]: codeActionPathList

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs [hunk 3]
```diff
@@ -222,22 +224,15 @@ private static bool IsCodeActionNotSupportedByLSP(IUnifiedSuggestedAction sugges
...
+            ref int currentHighestSetNumber)
         {
-            if (!string.IsNullOrEmpty(currentTitle))
-            {
-                // Adding a delimiter for nested code actions, e.g. 'Suppress or Configure issues|Suppress IDEXXXX|in Source'
-                currentTitle += '|';
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:231]: excellent.

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs [hunk 4]
```diff
@@ -222,22 +224,15 @@ private static bool IsCodeActionNotSupportedByLSP(IUnifiedSuggestedAction sugges
...
-            }
-
             var codeAction = suggestedAction.OriginalCodeAction;
-            currentTitle += codeAction.Title;
 
             var diagnosticsForFix = GetApplicableDiagnostics(request.Context, suggestedAction);
-
             // Nested code actions' unique identifiers consist of: parent code action unique identifier + '|' + title of code action
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:233]: is this comment correct anymore?

**Comment by CyrusNajmabadi** [comment]
would be great to have an integration test.  but not blocking on that.

**Review by dibarbet — APPROVED** [review]

**Pull request merged**